spec.html

```diff
@@ -34532,8 +34534,203 @@ <h2>Syntax</h2>
           [+UnicodeMode] `-`
           CharacterClassEscape[?UnicodeMode]
           CharacterEscape[?UnicodeMode]
+
+        ClassContents ::
+          ClassUnion
+          ClassIntersection
+          ClassSubtraction
+
+        ClassUnion ::
+          ClassRange ClassUnion?
+          ClassOperand ClassUnion?
+
+        ClassIntersection ::
+          ClassOperand `&amp;&amp;` [lookahead != `&amp;`] ClassOperand
+          ClassIntersection `&amp;&amp;` [lookahead != `&amp;`] ClassOperand
+
+        ClassSubtraction ::
+          ClassOperand `--` ClassOperand
+          ClassSubtraction `--` ClassOperand
+
+        ClassRange ::
+          ClassCharacter `-` ClassCharacter
+      </emu-grammar>
+      <ul>
+        <li>
+          It is a Syntax Error if the CharacterValue of the first |ClassCharacter| is larger than the CharacterValue of the second |ClassCharacter|.
+        </li>
+      </ul>
+      <emu-grammar type="definition">
+        ClassOperand ::
+          ClassCharacter
+          ClassStrings
+          NestedClass
+
+        NestedClass ::
+          `[` [lookahead != `^`] ClassRanges[+UnicodeMode, +UnicodeSetsMode] `]`
+          `[` `^` ClassRanges[+UnicodeMode, +UnicodeSetsMode] `]`
+          `\` CharacterClassEscape[+UnicodeMode]
       </emu-grammar>
+      <emu-note>
+        <p>The first two lines here are equivalent to CharacterClass.</p>
+        <p style="background: yellow">[It is less confusing not having to back out of the new syntax, only to dive back in. And if we wanted to use CharacterClass here, we would need to check for differences in static &amp; runtime semantics.]</p>
+      </emu-note>
+      <emu-grammar type="definition">
+        ClassStrings ::
+          `\q{` ClassString MoreClassStrings? `}`
+
+        MoreClassStrings ::
+          `|` ClassString MoreClassStrings?
+
+        ClassString ::
+          [empty]
+          NonEmptyClassString
+
+        NonEmptyClassString ::
+          ClassCharacter NonEmptyClassString?
 
+        ClassCharacter ::
+          [lookahead &notin; ClassReservedDouble] SourceCharacter but not ClassSyntaxCharacter
+          `\` CharacterEscape[+UnicodeMode]
+          `\` ClassAllowEscaped
+          `\` `b`
+
+        ClassReservedDouble :: one of
+          `&amp;&amp;` `!!` `##`
+          `$$` `%%` `**`
+          `++` `,,` `..`
+          `::` `;;` `&lt;&lt;`
+          `==` `&gt;&gt;` `??`
+          `@@` `^^` `__`
+          `&grave;&grave;` `~~`
+      </emu-grammar>
+      <emu-note>
+        <ul style="background: yellow">
+          <li>
+            FYI: no need to list `--` `((` `))` `{{` `}}` `//` `||` because those characters can't be used as unescaped ClassCharacters anyway
+          </li>
+        </ul>
+      </emu-note>
+      <emu-grammar type="definition">
+        ClassSyntaxCharacter :: one of
+          `(` `)` `[` `]` `{` `}` `/` `-` `\` `|`
```

Comment:
A single ampersand has no syntactic meaning and is thus not required to be escaped.
It can be used as a literal character.
It is *allowed* to be escaped via what's now called ClassSetReservedPunctuator.

Reply:
I don't think I understand the distinction, wouldn't an isolated `{` or `|` in a set-compatible character class also lack syntactic meaning? I guess this folds into the broader conversation about escaping at https://github.com/tc39/ecma262/pull/2418#discussion_r836025429 .

Reply:
You are right, there is a little more to it. We had a lot of discussions about escaping and reserved syntax...
1. We decided to have the same escaping rules inside a ClassString as well as outside (but still in ClassContents). ClassString is simply a sequence of ClassSetCharacter. This means that we have to require escaping of both `|` and `}`.
2. It's confusing to require escaping `}` but not `{`.
3. We also want to reserve `{ }` for future extensions. (And for a while they were used for ClassStringDisjunction syntax without the `\q` prefix.)
   See slide 16 of our 2021-aug TC39 meeting slides: [Reserved syntax](https://docs.google.com/presentation/d/1foloLW13Elu0kslVsmD1hR_qZQBn8INcNpdWl0rlHrI/edit#slide=id.gce3325eb6c_1_17)

The yellow NOTE 3 just below this part also briefly explains why we require escaping for each character for which we do so.

Reply:
moved to https://github.com/tc39/proposal-regexp-set-notation/issues/59